src/components/Amap/Amap.tsx

```diff
@@ -0,0 +1,180 @@
+            });
+    }, []);
+
+    /**
+     * handleProcessData: Callback to handle and append new data from the child process to the output.
+     * It updates the state by appending the new data received to the existing output.
+     * @param {string} data - The data received from the child process.
+     */
+    const handleProcessData = useCallback((data: string) => {
+        setOutput((prevOutput) => prevOutput + "\n" + data); // Append new data to the previous output.
+    }, []);
+
+    /**
+     * handleProcessTermination: Callback to handle the termination of the child process.
+     * Once the process termination is handled, it clears the process PID reference and
+     * deactivates the loading overlay.
+     * @param {object} param - An object containing information about the process termination.
+     * @param {number} param.code - The exit code of the terminated process.
+     * @param {number} param.signal - The signal code indicating how the process was terminated.
+     */
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            // If the process was successful, display a success message.
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+
+                // If the process was terminated manually, display a termination message.
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+
+                // If the process was terminated with an error, display the exit and signal codes.
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+
+            // Clear the child process pid reference. There is no longer a valid process running.
+            setPid("");
+
+            // Cancel the loading overlay. The process has completed.
+            setLoading(false);
+        },
+        [handleProcessData] // Dependency on the handleProcessData callback
+    );
+
+    /**
+     * onSubmit: Asynchronous handler for the form submission event.
+     * It sets up and triggers the amap tool with the given parameters.
+     * Once the command is executed, the results or errors are displayed in the output.
+     *
+     * @param {FormValuesType} values - The form values, containing the target host, port(s), and additional options.
+     */
+    const onSubmit = async (values: FormValuesType) => {
+        // Activate loading state to indicate ongoing process
+        setLoading(true);
+
+        // Construct arguments for the amap command based on form input
+        const args = [values.target, "-bqv", values.port, values.options];
+
+        // Execute the amap command via helper method and handle its output or potential errors
+        CommandHelper.runCommandWithPkexec("amap", args, handleProcessData, handleProcessTermination)
```

Comment:
I am pretty sure AMAP can be run without elevated privileges:
![image](https://github.com/Hardhat-Enterprises/Deakin-Detonator-Toolkit/assets/35479648/e9422664-9e1e-4b39-a043-f5dcb9733d82)
Therefore, I recommend using runCommandGetPIDAndOutput instead of runCommandWithPkexec. 

- [ ] Use runCommandGetPIDAndOutput

Reply:
Fixed issue Thank you @michaeljpigott